group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -761,6 +777,31 @@ public ClassicGroup classicGroup(
         }
     }
 
+    public boolean validateOnlineUpgrade(ClassicGroup classicGroup) {
```

Comment:
Does it have to be public? Should we add some javadoc?

Reply:
private should be enough. Let me add the javadoc